### PR TITLE
rewrite clustering coefficients for speed

### PR DESCRIPTION
**Pull request by jpfairbanks**
I improved the clustering coefficients code to use a 2 pass algorithm. The updated code introduces an in place version and keeps the old interface intact. The ipynb notebook shows the results of 3 options I considered.

### CODE REVIEW

File: src/community/clustering.jl [hunk 1]
```diff
@@ -26,29 +26,52 @@ the maximum number of possible triangles, respectively, for each node in the lis
 
 This function is related to the local clustering coefficient `r` by ``r=\frac{a}{b}``.
 """
```
Comment (sbromberger) [inline src/community/clustering.jl]: I think this docstring needs to be moved to the actual `local_clustering()` function, not be above the mutable one.

File: benchmark/triangles.ipynb [hunk 2]
```diff
@@ -0,0 +1,367 @@
+{
```
Comment (sbromberger) [inline benchmark/triangles.ipynb]: I'm not sure I like the idea of notebooks in the `benchmark/` directory. Perhaps we can put them in docs and create a section for "miscellaneous notes"?

Comment (jpfairbanks) [inline benchmark/triangles.ipynb]: Yeah that's fine

**Comment by jpfairbanks** [comment]
I addressed your comments and then found an inefficiency in clustering coefficients

**Comment by sbromberger** [comment]
Thanks. As soon as tests are done, I'll merge. This looks great, and I *really* like the notebook showing performance improvements.

**Comment by jpfairbanks** [comment]
Thanks, I like the idea doc/notebooks for storing notebooks that demonstrate things. If we ever need to experimentally test out some alternatives we can use a notebook and that adds to permanent documentation.

**Comment by sbromberger** [comment]
you broke it :)

**Comment by sbromberger** [comment]
Gah. Code coverage!

**Pull request merged**